base/compiler/tfuncs.jl

```diff
@@ -1067,6 +1069,20 @@ function _fieldtype_tfunc(@nospecialize(s), exact::Bool, @nospecialize(name))
 end
 add_tfunc(fieldtype, 2, 3, fieldtype_tfunc, 0)
 
+function valid_type_param(T)
+    isvarargtype(T) && return false
```

Comment:
I don't think this case is possible here.

Reply:
This case is still wrong (it also was before, so it's not necessarily on this PR to fix it):
```
julia> Core.Compiler.apply_type_nothrow(Any[Core.Const(Ref),Core.Const(Vararg)],Type{<:Ref})
true
```

Reply:
> I don't think this case is possible here.

Maybe, but we may want to use this function from other places, so I thought it better to cover all the cases.